File: cuarray/samples/cpp/src/CuArray_get.cpp
#include "cuarray.h"
#include <iostream>
#include <random>

int main() {
    std::cout
            << "Running "
            << __FILE__
            << std::endl;

/* Creates a new float CuArray instance that will have 10 rows
 * and 10 columns*/
    CuArray<float> *cuArray = new CuArray<float>();
    int m = 10; /* Number of rows */
    int n = 10; /* Number of columns */
    cuArray->init(m,
                  n);

/* Fill the CuArray with random values */
    for (int i = 0; i < m; i++) {
        for (int j = 0; j < n; j++) {
            cuArray->set((float) rand() / (float) RAND_MAX,
                         i,
                         j);
        }
    }

/* As it's name implies, get(i, j) returns the value at the
 * specified position (i, j) in the CuArray. */

/* Use the get method to print the value at each position in the CuArray. */
    for (int i = 0; i < m; i++) {
        for (int j = 0; j < n; j++) {
            std::cout
                    << cuArray->get(i,
                                    j)
                    << " ";
        }
        std::cout
                << std::endl;
    }

/* Free the CuArray. */
    delete cuArray;
    return 0;
}

File: cuarray/samples/python/CuArray_get.py
import numpy as np

"""
Always precede CuArray with the data type
Here we are importing the CuArray float template 
"""
from cuarray import FloatCuArray

print("Running", __file__)

"""
Create a new float CuArray instance with 
10 rows and 10 columns
"""
float_cuarray = FloatCuArray()
float_cuarray.init(10, 10)

"""Fill the array with random values"""

for i in range(10):
    for j in range(10):
        val = np.random.random()
        float_cuarray.set(val, i, j)

"""Print the array"""
print(float_cuarray)